test/integration/websocket_integration_test.cc

```diff
@@ -421,6 +421,51 @@ TEST_P(WebsocketIntegrationTest, BidirectionalChunkedData) {
   response_headers.removeContentLength();
   performUpgrade(request_headers, response_headers);
 
+  // Client codec should not add TE or CL:0 headers to UPGRADE requests without content-length
+  ASSERT_TRUE(upstream_request_->headers().TransferEncoding() == nullptr);
```

Comment:
I think it's worth having a test in integration_test.cc or a clone of those, which looks at raw headers.  The codec would strip TE so this check is a functional no-op and I believe would pass without the second part of your change.  AFIK the body payload would also be untouched either way, because once we hit the upgrade path we direct-encode all data.